### PR TITLE
Add option to disable recursive traversal of symbolic target

### PR DESCRIPTION
**Pull request by cjgu**
### What
Implement a way to optionally disable recursive traversal of symbolic target as mentioned in https://github.com/SuperCuber/dotter/issues/67#issuecomment-851022618 and https://github.com/SuperCuber/dotter/issues/74#issuecomment-864546939

Allows to symlink the directory instead of each file inside the directory:
```
vim = { target="~/.vim", type="symbolic", recursive=false }
```

### Implementation details 
I'm not too happy with the destructing of the SymbilicTarget and would be happy to take suggestion on improving it.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -21,6 +21,7 @@ pub struct SymbolicTarget {
     pub owner: Option<UnixUser>,
     #[serde(rename = "if")]
     pub condition: Option<String>,
+    pub recursive: Option<bool>,
```
Comment (SuperCuber) [inline src/config.rs:24]: This should probably be a `bool`, and by default deserialize to `true` when it's missing. Look into `serde field attributes` to make that happen

File: src/config.rs [hunk 2]
```diff
@@ -481,6 +483,10 @@ fn expand_directory(source: &Path, target: FileTarget) -> Result<Files> {
         let mut map = Files::new();
         map.insert(source.into(), target);
         Ok(map)
+    } else if let FileTarget::Symbolic(SymbolicTarget{target:_, owner:_, condition: _, recursive: Some(false)}) = target {
```
Comment (SuperCuber) [inline src/config.rs:486]: This can be `if let FileTarget::Symbolic(SymbolicTarget { recursive: false, .. })` with the previous change

**Review by SuperCuber — CHANGES_REQUESTED** [review]
In addition to that I'm worried about edge cases like changing from recursive to non recursive and then deploying or changing from `{ target="a", recursive=false }` to `{ target="a/foo" }` - gotta make sure the code handles those properly, maybe even add unit tests for these cases.

**Comment by SuperCuber** [comment]
Depending on whether it "just works" with the edge cases I mentioned this PR could be basically done or really complex, fingers crossed 🤞

**Comment by cjgu** [comment]
Thank you for the quick review! I'll try testing those edge cases and try writing some tests for them, and do the other adjustments.

**Comment by dead10ck** [comment]
@cjgu are you still planning on continuing this work?

**Comment by cjgu** [comment]
Hey! Sorry for lack of updates here.

I did a bit of testing on how to implement it to support toggling the `recursive` on a directory but it was quite complicated and my need of this changed a bit. If you want to take this further you are very welcome to do so.

**Comment by SuperCuber** [comment]
Closed via #94
